diff --git a/dist/ExpensiMark.js b/dist/ExpensiMark.js
index c1652995d689c71d386218a4df98419e0116799a..3c612053f4febaf036eca54e90650466b960dcc8 100644
--- a/dist/ExpensiMark.js
+++ b/dist/ExpensiMark.js
@@ -1,4 +1,5 @@
 "use strict";
+"worklet";
 var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
     if (k2 === undefined) k2 = k;
     var desc = Object.getOwnPropertyDescriptor(m, k);
diff --git a/dist/str.js b/dist/str.js
index bda25fcf8fb0f94fe7a259bc6b534b0eef07621e..7a44ab47c01f124f82cd8763862f204e98433485 100644
--- a/dist/str.js
+++ b/dist/str.js
@@ -1,4 +1,5 @@
 "use strict";
+"worklet";
 var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
     if (k2 === undefined) k2 = k;
     var desc = Object.getOwnPropertyDescriptor(m, k);
@@ -37,6 +38,7 @@ function resultFn(parameter, ...args) {
     return parameter;
 }
 const Str = {
+    __workletObject: true,
     /**
      * Return true if the string is ending with the provided suffix
      *
diff --git a/dist/utils.js b/dist/utils.js
index 70bc31f0dce2bdc72ddc475e7a22da12144d9f2b..38ee61f79fd9e67a4ff66d36c6b4628404d90af0 100644
--- a/dist/utils.js
+++ b/dist/utils.js
@@ -1,4 +1,5 @@
 "use strict";
+"worklet";
 Object.defineProperty(exports, "__esModule", { value: true });
 exports.isObject = exports.isFunction = exports.unescape = exports.escape = exports.isNavigatorAvailable = exports.isWindowAvailable = void 0;
 /** Checks if the `window` global object is available. */
